enterprise/backend/test/metabase_enterprise/serialization/v2/entity_ids_test.clj

```diff
@@ -45,24 +48,36 @@
                      (entity-id))))))))))
 
 (deftest drop-entity-ids-test
-  (testing "With a temp Collection with an entity ID"
-    (let [now (LocalDateTime/of 2022 9 1 12 34 56)]
-      (mt/with-temp! [Collection c {:name       "No Entity ID Collection"
-                                    :slug       "no_entity_id_collection"
-                                    :created_at now}]
-        (letfn [(entity-id []
-                  (some-> (t2/select-one-fn :entity_id Collection :id (:id c)) str/trim))]
-          (is (some? (entity-id)))
-          (testing "Should return truthy on success"
-            (is (= true
-                   (v2.entity-ids/drop-entity-ids!))))
-          (is (nil? (entity-id)))))))
-  (testing "empty table"
-    (testing "has no entity ids"
-      (mt/with-temp! [Collection _ {:name       "No Entity ID Collection"
-                                    :slug       "no_entity_id_collection"}]
-        (is (nil? (t2/select-fn-set :entity-id Dashboard)))
-        (testing "but doesn't crash drop-entity-ids"
-          (is (= true
-                 (v2.entity-ids/drop-entity-ids!)))
-          (is (nil? (t2/select-fn-set :entity-id Dashboard))))))))
+  (mt/with-empty-h2-app-db
+    (testing "With a temp Collection with an entity ID"
+      (let [now (LocalDateTime/of 2022 9 1 12 34 56)]
+        (mt/with-temp! [Collection c {:name       "No Entity ID Collection"
+                                      :slug       "no_entity_id_collection"
+                                      :created_at now}]
+                       (letfn [(entity-id []
+                                 (some-> (t2/select-one-fn :entity_id Collection :id (:id c)) str/trim))]
+                         (is (some? (entity-id)))
+                         (testing "Should return truthy on success"
+                           (is (= true
+                                  (v2.entity-ids/drop-entity-ids!))))
+                         (is (nil? (entity-id)))))))
+    (testing "empty table"
+      (testing "has no entity ids"
+        (mt/with-temp! [Collection _ {:name "No Entity ID Collection"
+                                      :slug "no_entity_id_collection"}]
+                       (is (nil? (t2/select-fn-set :entity-id Dashboard)))
+                       (testing "but doesn't crash drop-entity-ids"
+                         (is (= true
+                                (v2.entity-ids/drop-entity-ids!)))
+                         (is (nil? (t2/select-fn-set :entity-id Dashboard)))))))
+    (testing "entity_id field is nullable everywhere so that drop-entity-ids will work for"
```

Comment:
```suggestion
    (testing "entity_id field is nullable everywhere so that drop-entity-ids will work for"
```

I'd make this a `deftest` on its own, it doesn't really need to be run in an empty h2 db tho.

Reply:
and what if we make this testing context more actionable?

like `entity_id field should be nullable for model so that drop-entity-ids work (#36365)`

Reply:
It'll fail on pgsql since table name is not upper-case there. So I either need to dispatch logic on a db type, or bind it to a specific db. Later seems easier and less cluttered, so I went that way.

Why the whole test is wrapped in `empty-h2-db`: it reliably fails without it locally without it, and I couldn't understand why.

Reply:
Nevermind, code is ok either way 😁